assets/app/views/deployments.html

```diff
@@ -10,14 +10,112 @@
+                <copy-to-clipboard-button clipboard-text="'osc deploy ' + deploymentConfigName + ' -n ' + project.metadata.name"></copy-to-clipboard-button>
+              </dd>
+
+            </dl>
+          </div>
+          
+          <div class="well" style="margin-bottom: 10px;" ng-repeat="deployment in deploymentsByDeploymentConfig[deploymentConfigName] | orderObjectsByDate : true">
+            
+            <h3>
+              Deployment #{{deployment.metadata.annotations.deploymentVersion}}
+              <span ng-if="deployment.metadata.annotations.deploymentVersion == deploymentConfig.latestVersion">(latest)</span>
+            </h3>
+            <div>Name: {{deployment.metadata.name}}</div>
+            <div>Created: <relative-timestamp timestamp="deployment.metadata.creationTimestamp"></relative-timestamp></div>            
+            <div>
+              <span>Status: {{deployment.metadata.annotations.deploymentStatus}}</span>
+              <span ng-switch="deployment.metadata.annotations.deploymentStatus" class="hide-ng-leave">
+                <span ng-switch-when="Running"> for <duration-until-now timestamp="deployment.metadata.creationTimestamp"></duration-until-now></span>
+                <span ng-switch-when="New"> for <duration-until-now timestamp="deployment.metadata.creationTimestamp"></duration-until-now></span>
+                <span ng-switch-when="Pending"> for <duration-until-now timestamp="deployment.metadata.creationTimestamp"></duration-until-now></span>
+              </span>
+              <span ng-switch="deployment.metadata.annotations.deploymentStatus" class="hide-ng-leave">
+                <span ng-switch-when="Complete" class="fa fa-check text-success" aria-hidden="true"></span>
+                <span ng-switch-when="Failed" class="fa fa-times text-danger" aria-hidden="true"></span>
+                <span style="margin-left: 5px;" ng-switch-default class="fa fa-refresh fa-spin" aria-hidden="true"></span>
+              </span>
+            </div>
+            <div>Replicas: {{deployment.spec.replicas}}</div>
+            <pod-template ng-init="podTemplate = deployment.spec.template"></pod-template>
+          </div>
+        </div>
+      </div>
+
+      <!-- render any deployments whose deployment configs no longer exist -->
+      <div class="tile" ng-repeat="(deploymentConfigName, deployments) in deploymentsByDeploymentConfig" ng-if="!deploymentConfigs[deploymentConfigName]">
+        <h2 ng-if="deploymentConfigName != ''">
+          {{deploymentConfigName}}
+          <span class="pficon-layered" data-toggle="tooltip" data-placement="right" title="This deployment config no longer exists" style="cursor: help;">
+            <span class="pficon pficon-warning-triangle"></span>
+            <span class="pficon pficon-warning-exclamation"></span>
+          </span>            
+        </h2>
+        <div style="margin-bottom: 10px;" ng-repeat="deployment in deployments | orderObjectsByDate : true">
```

Comment:
add ` track by (deployment | uid)` to the end of the `ng-repeat` attribute... lets angular track objects so that if their status updates, it can match the object to the already-rendered html and just update what's changed